### ISSUE TITLE
Deploy is successful, but out script exits with error

### ISSUE THREAD

**Issue by shinmyung0** [patrickcrocker/maven-resource#10]
So I am currently using this resource to deploy to a internal snapshot repo. The resource successfully deploys everything, but the script still exits with an error code which causes the build to fail:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.374 s
[INFO] Finished at: 2017-09-26T00:26:19Z
[INFO] Final Memory: 12M/432M
[INFO] ------------------------------------------------------------------------
jq - commandline JSON processor [version 1.5]
Usage: jq [options] <jq filter> [file...]

	jq is a tool for processing JSON inputs, applying the
	given filter to its JSON text inputs and producing the
	filter's results as JSON on standard output.
	The simplest filter is ., which is the identity filter,
	copying jq's input to its output unmodified (except for
	formatting).
	For more advanced filters see the jq(1) manpage ("man jq")
	and/or https://stedolan.github.io/jq

	Some of the options include:
	 -c		compact instead of pretty-printed output;
	 -n		use `null` as the single input value;
	 -e		set the exit status code based on the output;
	 -s		read (slurp) all inputs into an array; apply filter to it;
	 -r		output raw strings, not JSON texts;
	 -R		read raw strings, not JSON texts;
	 -C		colorize JSON;
	 -M		monochrome (don't colorize JSON);
	 -S		sort keys of objects on output;
	 --tab	use tabs for indentation;
	 --arg a v	set variable $a to value <v>;
	 --argjson a v	set variable $a to JSON value <v>;
	 --slurpfile a f	set variable $a to an array of JSON texts read from <f>;
	See the manpage for more options.
```
As you can see from the build output, it seems to be some issue with how `jq` is being called.

**Comment by patrickcrocker** [comment]
Can you post a sanitized version of your pipeline?

**Comment by shinmyung0** [comment]
Sure thing. 
```
---
resource_types:
- name: maven-resource
  type: docker-image
  source:
    repository: pivotalpa/maven-resource
    tag: latest


resources:
- name: catalog-snapshot
  type: maven-resource
  source:
    url: {{nexus-release-url}}
    snapshot_url: {{nexus-snapshot-url}}
    artifact: com.tixhub.catalog:catalog-service:jar
    username: {{stub-username}}
    password: {{stub-password}}
    repository_cert: {{mvn-repo-cert}}

- name: catalog-service-code
  type: git
  source:
    uri: {{catalog-repo-uri}}
    private_key: {{github-private-key}}
    ignore_paths:
      - .gitignore

- name: cf-push-catalog
  type: cf
  source:
    api: {{cf-api}}
    username: {{cf-username}}
    password: {{cf-password}}
    organization: {{cf-org}}
    space: {{cf-space}}
    skip_cert_check: true

jobs:
- name: build-test-docs
  serial: true
  plan:
  - get: catalog-service-code
    trigger: true

  - task: install
    config:
      platform: linux
      image_resource:
        type: docker-image
        source:
          repository: corp.docker.repo:443/jdk8
          tag: "latest"
          username: {{stub-username}}
          password: {{stub-password}}
          insecure_registries:
            - "corp.docker.repo:443"
      inputs:
        - name: catalog-service-code
      outputs:
        - name: build-output
      caches:
        - path: catalog-service-code/.m3
      run:
        dir: catalog-service-code
        path: ./pipeline/build.sh
  - put: catalog-snapshot
    params:
      file: build-output/target/catalog-service.jar
      pom_file: catalog-service-code/pom.xml
      version_file: catalog-service-code/version
- name: test-deploy
  plan:
  - get: catalog-snapshot
    trigger: true
    passed:
      - build-test-docs
  - put: cf-push-catalog
    params:
      manifest: catalog-service-code/manifest.yml
      current_app_name: catalog-service
```

**Comment by pcrocker-pivotal** [comment]
You can use the undocumented `debug: true` parameter in your source config to output more info:

```yml
- name: catalog-snapshot
  type: maven-resource
  source:
    url: {{nexus-release-url}}
    snapshot_url: {{nexus-snapshot-url}}
    artifact: com.tixhub.catalog:catalog-service:jar
    username: {{stub-username}}
    password: {{stub-password}}
    repository_cert: {{mvn-repo-cert}}
    debug: true
```

WARNING: this will output all your variables (including your password) to the build output, so use with caution (make sure your pipeline is not exposed on a public facing Concourse).

What are you using for your Maven Repository Manager (Nexus 2.x, Nexus 3.x, Artifactory, etc)?

**Comment by shinmyung0** [comment]
So I figured out what the issue was. Inside of the `check` script there is `curl -s ` command which was failing silently due to certificate issues. This made the $version return as empty from the `check` script which triggered the help dialog for `jq`.  Setting `skip_cert_check: true` made everything worked.

I tried setting `debug: true` and it also does not show this issue. Could there be a clean way of handling errors inside of this variable assignment instead of using `curl -s`, which swallows up any potentially useful error messages.
